docs/demos/Video_Processing.py

```diff
@@ -0,0 +1,607 @@
+#   Soup tutorial on
+#   `using the Kalman Filter <https://stonesoup.readthedocs.io/en/latest/auto_tutorials/01_KalmanFilterTutorial.html>`_.
+#
+# Data Association
+# ****************
+# We utilise a :class:`~.DistanceHypothesiser` to generate hypotheses between tracks and
+# measurements, where :class:`~.Mahalanobis` distance is used as a measure of quality:
+from stonesoup.hypothesiser.distance import DistanceHypothesiser
+from stonesoup.measures import Mahalanobis
+hypothesiser = DistanceHypothesiser(predictor, updater, Mahalanobis(), 10)
+# %%
+# Continuing the :class:`~.GNNWith2DAssigment` class is used to perform fast joint data association,
+# based on the Global Nearest Neighbour (GNN) algorithm:
+from stonesoup.dataassociator.neighbour import GNNWith2DAssignment
+data_associator = GNNWith2DAssignment(hypothesiser)
+#%%
+# .. note::
+#   For more information on the above classes and how they operate you can refer to the
+#   `Data Association - clutter <https://stonesoup.readthedocs.io/en/latest/auto_tutorials/05_DataAssociation-Clutter.html>`_
+#   and `Data Association - Multi-Target Tracking <https://stonesoup.readthedocs.io/en/latest/auto_tutorials/06_DataAssociation-MultiTargetTutorial.html>`_
+#   tutorials.
+
+# %%
+# Track Initiation
+# ****************
+# For initialising tracks we will use a :class:`~.MultiMeasurementInitiator`, which allows our
+# tracker to tentatively initiate tracks from unassociated measurements, and hold them within the
+# initiator until they have survived for at least 10 frames. We also define a
+# :class:`~.UpdateTimeStepsDeleter` deleter to be used by the initiator to delete tentative tracks
+# that have not been associated to a measurement in the last 3 frames.
+from stonesoup.types.state import GaussianState
+from stonesoup.types.array import CovarianceMatrix, StateVector
+from stonesoup.initiator.simple import MultiMeasurementInitiator
+from stonesoup.deleter.time import UpdateTimeStepsDeleter
+prior_state = GaussianState(StateVector(np.zeros((6,1))),
+                            CovarianceMatrix(np.diag([100**2, 30**2, 100**2, 30**2, 100**2, 100**2])))
+deleter_init = UpdateTimeStepsDeleter(time_steps_since_update=3)
+initiator = MultiMeasurementInitiator(prior_state, measurement_model, deleter_init,
+                                      data_associator, updater, min_points=10)
+
+# %%
+# Track Deletion
+# **************
+# For confirmed tracks we used again a :class:`~.UpdateTimeStepsDeleter`, but this time configured
+# to delete tracks after they have not bee associated to a measurement in the last 15 frames.
+deleter = UpdateTimeStepsDeleter(time_steps_since_update=15)
+#%%
+# .. note::
+#
+#   For more information on the above classes and how they operate you can refer to the Stone
+#   `Initiators & Deleters <https://stonesoup.readthedocs.io/en/latest/auto_tutorials/09_Initiators_&_Deleters.html>`_
+#   tutorial.
+#
```

Comment:
Think this # needs to be blank line, as causes heading below to break.

Reply:
@sdhiscocks Thank you Steve, I have fixed that in the latest version.